src/Stormpath/Resource/Application.php

```diff
@@ -355,6 +356,11 @@ public function handleIdSiteCallback($responseUri)
 
         $jwt = JWT::decode($token, $apiSecret, array('HS256'));
 
+        if(isset($jwt->err)) {
+            $error = json_decode($jwt->err);
+            throw new IDSiteException($error->developerMessage, $error->code);
```

Comment:
If errors are now coming in the same format of a `ResourceError`, why don't we construct and throw a `ResourceError` instead of throwing this new exception with less information than a `ResourceError` would have?

Also, according to the specs it looks like we should be doing something for a specific type of error from the SDK (not 100% sure).